services/src/test/java/org/broadleaf/payment/service/gateway/PayPalCheckoutTransactionServiceIT.java

```diff
@@ -0,0 +1,546 @@
+import org.broadleafcommerce.payment.service.gateway.PayPalCheckoutPaymentGatewayType;
+import org.broadleafcommerce.payment.service.gateway.PayPalCheckoutRestConfigurationProperties;
+import org.broadleafcommerce.payment.service.gateway.PayPalCheckoutTransactionService;
+import org.broadleafcommerce.vendor.paypal.service.payment.MessageConstants;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalAuthorizationRetrievalRequest;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalAuthorizationRetrievalResponse;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalAuthorizeRequest;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalAuthorizeResponse;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalCaptureRequest;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalCaptureResponse;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalCaptureRetrievalRequest;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalCaptureRetrievalResponse;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalRefundRequest;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalRefundResponse;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalSaleRequest;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalSaleResponse;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalSaleRetrievalRequest;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalSaleRetrievalResponse;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalVoidRequest;
+import org.broadleafcommerce.vendor.paypal.service.payment.PayPalVoidResponse;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+import com.broadleafcommerce.money.util.MonetaryUtils;
+import com.broadleafcommerce.paymentgateway.domain.PaymentRequest;
+import com.broadleafcommerce.paymentgateway.domain.PaymentResponse;
+import com.broadleafcommerce.paymentgateway.domain.enums.DefaultTransactionFailureTypes;
+import com.broadleafcommerce.paymentgateway.domain.enums.DefaultTransactionTypes;
+import com.broadleafcommerce.paymentgateway.service.exception.PaymentException;
+import com.paypal.api.payments.Authorization;
+import com.paypal.api.payments.Capture;
+import com.paypal.api.payments.DetailedRefund;
+import com.paypal.api.payments.Error;
+import com.paypal.api.payments.Payment;
+import com.paypal.api.payments.RelatedResources;
+import com.paypal.api.payments.Sale;
+import com.paypal.api.payments.Transaction;
+import com.paypal.base.rest.PayPalRESTException;
+
+import java.util.Collections;
+
+/**
+ * @author Dima Myroniuk (dmyroniuk)
+ */
+@SpringBootTest
+public class PayPalCheckoutTransactionServiceIT {
```

Comment:
If we're going to use mocks this much, then this should just be a unit test.

How realistic would it be to call the actual PayPal APIs via integration tests? That's how we'll get a much better sense of whether the integration is passing or not.

Reply:
Calling the actual PayPal APIs can cause issues sometimes and these tests should be disabled by default. Also, we will need to provide the `clientId` and `clientSecret` to execute the requests.

Reply:
Agreed. In APS & Checkout.com, we've made use of the following annotation to disable the tests by default, but allow us to execute the suite if we provide a property.

```
@EnabledIf(
        expression = "#{'true'==systemProperties['broadleaf.checkout-com.live-regression-tests.enabled']}",
        reason = "Regression tests hitting live Checkout.com endpoints must be conditionally enabled")
```

Reply:
Looks like it is not possible to execute the payment without approved payment. I couldn't find how to mock the approval flow from the tests.

Reply:
We found that producing a PayerId isn't realistic for a backend IT & that it requires an integration with the frontend library. Therefore, we decided to move forward without the ITs.